### PR TITLE
Add pre-generated StreamingAEAD key templates

### PR DESCRIPTION
**Pull request by pkorth**
Allows client code to construct a StreamingAead KeyTemplate for use with StreamingAeadFactory.

### CODE REVIEW

File: java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ *       KeysetHandle.generateNew(StreamingAeadKeyTemplates.AES128_GCM_HKDF_STREAMING);
+ *   StreamingAead streamingAead = StreamingAeadFactory.getPrimitive(handle);
+ * </pre>
+ */
+public final class StreamingAeadKeyTemplates {
+
+  /**
+   * A {@code KeyTemplate} that generates new instances of {@code AesGcmHkdfStreamingKey} with the
+   * following parameters:
+   *   - AES key size: 16 bytes
```
Comment (thaidn) [inline java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java]: Nit: could you please change this to
<ul>
  <li>AES key size: 16 bytes</li>
  ...
</ul>

We're going to change the other template files.

File: java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+   *   - AES IV size: 16 bytes
+   *   - HMAC tag size: 16 bytes
+   *   - Ciphertext segment size: 256
```
Comment (thaidn) [inline java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java]: 256 is a bit too small. I think we should use 4KB, which is usually the size of a memory page in most operating systems.

**Comment by thaidn** [comment]
@przydatek @bleichen FYI

File: java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java (hunk 1 above)
Comment (pkorth) [reply java/src/main/java/com/google/crypto/tink/streamingaead/StreamingAeadKeyTemplates.java]: Your comment got markdown-formatted so it's unclear what you originally typed -- do you mean remove the hyphen and indentation, like "* AES key size: 16 bytes"?

**Pull request merged**